src/pages/index.js

```diff
@@ -37,44 +37,43 @@ const userInfo = new UserInfo(
   "#profile-des",
   "#profile-image"
 );
-let userID;
-api
-  .getUserinfo()
-  .then((data) => {
-    const { name, about, avatar, _id } = data;
-    userID = _id;
-    userInfo.setUserInfo(name, about);
-    userInfo.setUserProfileImage(avatar);
-  })
-  .catch((error) => {
-    console.error("Error fetching data:", error);
-  });
+
+let userId;
+let cardSection;
 
 // functions
 function handleProfileFormSubmit(data) {
   const { name, about } = data;
   api
     .updateProfile(name, about)
-    .then(() => {
-      profileEditPopup.editSubmitBtn();
-    })
+    .then(profileEditPopup.editSubmitBtn())
     .then(() => {
       userInfo.setUserInfo(name, about);
-      profileEditPopup.close();
-      profileEditPopup.resetSubmitBtn();
     })
-    .catch((error) => {
-      console.error("Error fetching data:", error);
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditPopup.resetSubmitBtn();
+      profileEditPopup.close();
```

Comment:
`      profileEditPopup.close();
`

You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well



**Please, correct it everywhere**